app/controllers/messages_controller.rb

```diff
@@ -0,0 +1,41 @@
+class MessagesController < ApplicationController
+  before_action :require_login
+  before_action :require_leader, except: :create
+  before_action :require_leader_or_rvt_member, only: :create
+  before_action :require_application_or_evaluation_season
+
+  def index
+    @semester = SystemAttribute.semester.to_s
+    @group = Group.find(params[:group_id])
```

Comment:
It's little like the current_group in application_controller.
I think we should use here and below.